.github/workflows/tests.yml

```diff
@@ -20,13 +20,9 @@ jobs:
       - run: scripts-dev/config-lint.sh
 
   lint:
-    # This does a vanilla `poetry install` - no extras. I'm slightly anxious
-    # that we might skip some typechecks on code that uses extras. However,
-    # I think the right way to fix this is to mark any extras needed for
-    # typechecking as development dependencies. To detect this, we ought to
-    # turn up mypy's strictness: disallow unknown imports and be accept fewer
-    # uses of `Any`.
-    uses: "matrix-org/backend-meta/.github/workflows/python-poetry-ci.yml@v1"
+    uses: "matrix-org/backend-meta/.github/workflows/python-poetry-ci.yml@dmr/extras-for-typechecking"
+    with:
+      typechecking-extras: "all"
```

Comment:
Looks like I was right to be anxious.

On reflection, I think the approach I described on the left was too idealistic.

Before merging the branch of `backend-meta` should be reset to point to the v1 tag.